ios/brave-ios/Sources/BraveWallet/Crypto/Stores/SendTokenStore.swift

```diff
@@ -865,6 +937,41 @@ public class SendTokenStore: ObservableObject, WalletObserverStore {
     }
   }
 
+  private func sendTokenOnBtc(
+    amount: String,
+    token: BraveWallet.BlockchainToken,
+    from fromAccountId: BraveWallet.AccountId,
+    completion: @escaping (_ success: Bool, _ errMsg: String?) -> Void
+  ) {
+    isMakingTx = true
+    guard
+      let amountInSatoshi = WeiFormatter.decimalToAmount(
+        amount.normalizedDecimals,
+        tokenDecimals: Int(token.decimals)
+      )
+    else {
+      isMakingTx = false
```

Comment:
All these places which have a guard at the top you may as well not set `isMakingTx` to `true` until after it passes

Reply:
Hmm, part if it is that `sendToken(amount:completion:)` now also assigns `isMakingTx = true` before async request to fetch all accounts. I'm not sure if it makes sense to set to false in `sendToken(amount:completion:)` before calling another function (ex. this `sendTokenOnBtc(amount:token:from:completion:)`) re-assigns true as `isMakingTx` is a `@Published` property triggering SwiftUI body update... though re-assignment from `true` -> `true` also does, but SwiftUI heuristics _should_ know not to update in that case.